### ISSUE TITLE
Unable to Build With gradle build

### ISSUE THREAD

**Issue by DivyanshByte** [StartForKillerMC/JavaMinecraft#3]
How to Build This Program with gradle? i am unable to do it.

**Comment by StartForKiller** [comment]
Then share the error that the build gives

**Comment by DivyanshByte** [comment]
I tried building the app using `gradle build`
but i am getting no jar files in build folder except in the libs folder there is a Minecraft-1.0.0_Snapshot.jar
So How to get the Jar file

**Comment by samuelh2005** [comment]
That is the jar

**Comment by samuelh2005** [comment]
Thats how gradle works

**Comment by DivyanshByte** [comment]
No it is not running and giving error no manifest found

**Comment by samuelh2005** [comment]
Then the gradle file is wrong,

**Comment by samuelh2005** [comment]
the mainclass is not being set

**Comment by DivyanshByte** [comment]
So Can You Give me the product file made afterwards or you can convert this to maven project or pom.xml bcoz i tried to change some things in the gradle but it don't work

**Comment by samuelh2005** [comment]
add this to the bottom of build.gradle
```
mainClassName = "io.github.startforkiller.jminecraft.game.Main"

project.configurations.implementation.setCanBeResolved(true)
jar {
    manifest {
        attributes(
                "Main-Class": "$mainClassName",
                "Class-Path":  configurations.implementation.collect { it.name }.join(' ')
        )

    }

    dependsOn ('dependencies')

    from {
        configurations.runtimeClasspath.collect { it.isDirectory() ? it : zipTree(it) }
    }
}
```
And then run using the jar task insted

**Comment by DivyanshByte** [comment]
Ok Thanks You Can Also Add This in the repository as a new commit to help new ones Closing This Now. :)

**Comment by StartForKiller** [comment]
Yeah, i haven't put the mainclass on the gradle build, because i always work with intellij that runs without it, i should add it

**Comment by DivyanshByte** [comment]
So How Can I open It In Intellij There it was also not opening

**Comment by DivyanshByte** [comment]
This is Not Even Working Now With your changes it's giving Could not set unknown property 'mainClassName' for root project 'Minecraft' of type org.gradle.api.Project.

**Comment by samuelh2005** [comment]
And replace the top bit with
```
plugins {
    id 'java'
    id 'application'
}
```

**Comment by DivyanshByte** [comment]
It's Now Giving me This error `Entry META-INF/INDEX.LIST is a duplicate but no duplicate handling strategy has been set.`

**Comment by samuelh2005** [comment]
Make sure you dont have 2 plugin blocks at the top

**Comment by DivyanshByte** [comment]
What? Can you explain?

**Comment by DivyanshByte** [comment]
And Also I am on Macos

**Comment by DivyanshByte** [comment]
ok wait I Have only one plugin block

**Comment by samuelh2005** [comment]
im on mac to

**Comment by DivyanshByte** [comment]
How do you made it work?

**Comment by samuelh2005** [comment]
if you want to see my [build.gradle](https://github.com/ajh123-development/HistorySurvival-Java-/blob/main/build.gradlel) however it is a different project, but you can stitch the two build.gradles togther

**Comment by DivyanshByte** [comment]
Ok You Can Give It to me

**Comment by samuelh2005** [comment]
its the link

**Comment by samuelh2005** [comment]
[try this one](https://github.com/ajh123-development/HistorySurvival-Java-/blob/main/build.gradle)

**Comment by DivyanshByte** [comment]
Ok

**Comment by samuelh2005** [comment]
and run the jar task, since i just use an IDE i can just press the button to do so

**Comment by DivyanshByte** [comment]
No it is still giving the same error

**Comment by samuelh2005** [comment]
dont know then :(

**Comment by DivyanshByte** [comment]
You Can give me the build jar to test it

**Comment by samuelh2005** [comment]
i have not built this one yet

**Comment by DivyanshByte** [comment]
so how are you testing it?

**Comment by samuelh2005** [comment]
Im not, just seeing what this one is missing

**Comment by DivyanshByte** [comment]
Oh Btw Thanks :)
